lib/astutils.cpp

```diff
@@ -2419,9 +2419,15 @@ static bool isExpressionChangedAt(const F& getExprTok,
     if (depth < 0)
         return true;
     if (tok->exprId() != exprid) {
-        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure()))
-            // TODO: Is global variable really changed by function call?
-            return true;
+        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure())) {
+            const Token* exprTok = getExprTok();
+            if (!((exprTok && exprTok->variable() && exprTok->variable()->scope() && tok->function() && tok->function()->isConst() &&
```

Comment:
I'm not even sure that `globalvar == true` is correct in this case, since we are dealing with member variables. But maybe they are global-like in this context? @pfultz2

Reply:
Where is `globalvar` being set? It may be set because there isn't the alias analysis there it handle member variables properly.

Reply:
It comes from here:
https://github.com/danmar/cppcheck/blob/343d04feb4719a9ac2760639168fd29e72a5bfa4/lib/astutils.cpp#L2575

Reply:
I see, global should be probably be false for class variables.